### PR TITLE
Feature/play videos like reels

### PR DESCRIPTION
**Pull request by guruprasadDev**
Implementing reels on Instagram clone
reel_ui

### CODE REVIEW

File: app/src/main/java/com/guruthedev/instagram/AccountSettingsActivity.kt [hunk 1]
```diff
@@ -7,27 +7,37 @@ import androidx.databinding.DataBindingUtil
...
         super.onCreate(savedInstanceState)
-        binding = DataBindingUtil.setContentView(this,R.layout.activity_account_settings)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_account_settings)
         EventBus.getDefault().register(this)
         binding.addBtn.setOnClickListener {
-                val intent = Intent(this,PassingData::class.java)
-                intent.putExtra("full_name",binding.fullNameProfileFrag.text.toString())
-                intent.putExtra("username",binding.usernameProfileFrag.text.toString())
-                intent.putExtra("bio",binding.bioProfileFrag.text.toString())
+            val intent = Intent(this, PassingData::class.java)
```
Comment (subirMM) [inline app/src/main/java/com/guruthedev/instagram/AccountSettingsActivity.kt]: use scope function properly for intent and binding

File: app/src/main/res/layout/activity_account_settings.xml [hunk 2]
```diff
@@ -126,8 +153,8 @@
             android:layout_marginHorizontal="@dimen/small_20dp"
             android:layout_marginTop="@dimen/small_4dp"
             android:layout_marginBottom="@dimen/small_8dp"
-            android:backgroundTint="@color/colorPrimaryDark"
-            android:text="Add"
+            android:backgroundTint="@color/colorBlack"
+            android:text="@string/add_btn"
```
Comment (subirMM) [inline app/src/main/res/layout/activity_account_settings.xml]: please rename add_btn to something else

File: app/src/main/java/com/guruthedev/instagram/AccountSettingsActivity.kt [hunk 3]
```diff
@@ -7,27 +7,37 @@ import androidx.databinding.DataBindingUtil
...
                 startService(intent)
+            }
         }
     }
-    @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
-    fun onEvent(resultData: ResultData){
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onEvent(resultData: ResultData) {
+        binding.apply {
```
Comment (subirMM) [inline app/src/main/java/com/guruthedev/instagram/AccountSettingsActivity.kt]: can we try using scope function properly for resultData and binding

File: app/src/main/java/com/guruthedev/instagram/ui/fragments/ProfileFragment.kt [hunk 4]
```diff
@@ -27,7 +27,9 @@ class ProfileFragment : Fragment() {
         binding.fullNameProfileFrag.text = fullName.toString()
         binding.profileFragmentUsername.text = username.toString()
         binding.bioProfileFrag.text = bio.toString()
+        binding.postImages.setOnClickListener {
 
+        }
         return binding.root
     }
 }
```
Comment (mm-aman) [inline app/src/main/java/com/guruthedev/instagram/ui/fragments/ProfileFragment.kt:35]: I see "No new line at end of file" warning with many files. Please add new line.

File: app/src/main/java/com/guruthedev/instagram/utils/VerifyInternetConnection.kt [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    override fun onInactive() {
+        connectivityManager.unregisterNetworkCallback(networkCallback)
+    }
+
+    private fun createNetworkCallback() = object : ConnectivityManager.NetworkCallback() {
+
+        override fun onAvailable(network: Network) {
+            Log.d(TAG, "onAvailable: $network")
```
Comment (mm-aman) [inline app/src/main/java/com/guruthedev/instagram/utils/VerifyInternetConnection.kt:44]: remove logs before pushing code

File: app/src/main/java/com/guruthedev/instagram/utils/VerifyInternetConnection.kt [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            val networkCapabilities = connectivityManager.getNetworkCapabilities(network)
+            val hasInternetCapability = networkCapabilities?.hasCapability(NET_CAPABILITY_INTERNET)
+            Log.d(TAG, "onAvailable: ${network}, $hasInternetCapability")
+
+            if (hasInternetCapability == true) {
+                // Check if this network actually has internet
+                CoroutineScope(Dispatchers.IO).launch {
```
Comment (mm-aman) [inline app/src/main/java/com/guruthedev/instagram/utils/VerifyInternetConnection.kt:51]: do we really need a coroutine scope here ?

File: app/src/main/res/layout/video_list_item.xml [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <VideoView
+            android:id="@+id/video_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
```
Comment (mm-aman) [inline app/src/main/res/layout/video_list_item.xml]: avoid using **constraintHorizontal_bias](app:layout_constraintHorizontal_bias="0.0"**

File: app/src/main/java/com/guruthedev/instagram/ui/fragments/PostFragment.kt [hunk 8]
```diff
@@ -40,21 +40,21 @@ class PostFragment : Fragment() {
...
-                            Snackbar.LENGTH_INDEFINITE
-                    ).show()
-                }
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { permissionGranted ->
+            if (permissionGranted) {
+                startCamera()
+            } else {
+                Snackbar.make(
+                    binding.root,
+                    "The camera permission is necessary",
```
Comment (mm-aman) [inline app/src/main/java/com/guruthedev/instagram/ui/fragments/PostFragment.kt]: please move this string to string.xml

**Review by mm-aman — CHANGES_REQUESTED** [review]
Pointed few minor issues. Please look out on them. 
Overall looks good to me.

**Review by subirMM — APPROVED** [review]

**Pull request merged**